pkg/network/dhcp/dhcp.go

```diff
@@ -41,25 +42,47 @@ func (p *ProbeOutput) IPNet() (*net.IPNet, error) {
 }
 
 func Probe(ctx context.Context, inf string) (output ProbeOutput, err error) {
-	// use udhcpc to prope the interface.
-	// this depends on that the interface is UP
-	cmd := exec.CommandContext(ctx, "udhcpc",
-		"-i", inf, //the interface to prope
-		"-q",       // exist once lease is optained
-		"-f",       //foreground
-		"-t", "20", //send 20 dhcp queries
-		"-T", "1", // every second
-		"-s", "/usr/share/udhcp/probe.script", // use the prope script
-		"--now", // exit if lease is not obtained
-	)
+	// use udhcpc to probe the interface.
+	// this check that the interface is UP before running udhcpc
+
+	interfaceByName, err := net.InterfaceByName(inf)
+	if err != nil {
+		return output, err
+	}
+
+	if (interfaceByName.Flags & net.FlagUp) == 0 {
+		return output, fmt.Errorf("failed to probe interface %s interface is not up", inf)
+	}
 
 	var stdout bytes.Buffer
 	var stderr bytes.Buffer
 
-	cmd.Stdout = &stdout
-	cmd.Stderr = &stderr
-	if err := cmd.Run(); err != nil {
-		return output, errors.Wrapf(err, "failed to prope interface '%s': %s", inf, stderr.String())
+	t := 10
+	var maxRetries uint64 = 5
+	check := func() error {
+		args := []string{
+			"-i", inf, // the interface to prope
+			"-q",      // exit once lease is optained
+			"-f",      // foreground
+			"-T", "1", // pauses for a second between packets timeout
+			"-t", fmt.Sprint(t), // send 't' dhcp discover packets
+			"-s", "/usr/share/udhcp/probe.script", // use the probe script
+			"--now", // exit if lease is not obtained
+		}
+
+		cmd := exec.CommandContext(ctx, "udhcpc", args...)
+		cmd.Stdout = &stdout
+		cmd.Stderr = &stderr
+
+		t += 5
+		return cmd.Run()
+	}
+
+	if err := backoff.Retry(
+		check,
+		backoff.WithMaxRetries(backoff.NewExponentialBackOff(), maxRetries),
+	); err != nil {
+		return output, errors.Wrapf(err, "failed to probe interface '%s': %s", inf, stderr.String())
```

Comment:
The udhcpc command already retries and you can control how many times and delay between sends via it's argument (the `-t` and `-T`) options I am wondering if it's necessary to implement another retry mechanism on top